### ISSUE TITLE
[FEA] Implement `nbytes` attribute on DeviceBuffer objecct

### ISSUE THREAD

**Issue by quasiben** [rapidsai/rmm#185]
`obj.nbytes` is a commonly used attributed in both numba and cupy and is also used extensively inside of dask.  

# example definition
```python
def nbytes(self):
    retrun self.dtype.itemsize * self.size
```

In considering a switch to using DeviceBuffer in dask rmm should have nbytes defined

cc @jakirkham

**Comment by jakirkham** [comment]
Yeah that sounds reasonable. Is this a nice to have or are you blocked by this?

**Comment by jakirkham** [comment]
Also minor note: For `DeviceBuffer` the `itemsize` is `1`.

**Comment by shwina** [comment]
This is just going to be an alias for `obj.size` as `DeviceBuffer` is not meant to be typed. Do we still need it? `DeviceBuffer` is really not meant to be treated as an array-like object similar to numpy/cupy arrays.

**Comment by quasiben** [comment]
If this is going to be used in dask, the `nbytes` attribute is a blocker for usage:

https://github.com/dask/distributed/blob/a285267d3c4042cc3aada675d6dabb793e43c500/distributed/comm/ucx.py#L146

https://github.com/dask/distributed/blob/a285267d3c4042cc3aada675d6dabb793e43c500/distributed/utils.py#L1113

**Comment by shwina** [comment]
Cool. I have no objection to adding an alias

**Comment by jakirkham** [comment]
Do you want to do the honors @shwina? 😉

**Comment by jakirkham** [comment]
Are there other issues like this Ben? 🙂

**Comment by quasiben** [comment]
I don't believe so.  I lightly tested dask/rmm/ucx with these additions and everything passed but if there any lingering issue I'll be sure to file a bug report
